modules/database/funcs.go

```diff
@@ -219,24 +250,28 @@ func (d *Database) getDBWithName(
 	h *helper.Helper,
 ) error {
 	db := &mariadbv1.MariaDBDatabase{}
+	namespace := d.namespace
+	if namespace == "" {
+		namespace = h.GetBeforeObject().GetNamespace()
```

Comment:
If the object `h` is wrapping is cluster-scoped, couldn't this set `namespace` to the empty string?  If so, would that cause problems for the `Get` on line 257 (where the namespace is needed)?

Reply:
I think the expected usage is:
* if h wraps cluster scoped object then provide a namespace explicitly. This way L257 will have a proper namespace.
* if h wraps a namespaced object then you don't need to provide a namespace explicitly and the call will use the same namespace as h.